### ISSUE TITLE
Incorrect error when specifying `impl Trait` explicitly

### ISSUE THREAD

**Issue by ChayimFriedman2** [rust-lang/rust#139140]
### Code

```Rust
struct S;
struct F;
impl F {
    fn foo(self, _x: impl Sized) {}
}

fn test() {
    F.foo::<u32>(S);
}
```

### Current output

```Shell
error[E0107]: method takes 0 generic arguments but 1 generic argument was supplied
 --> src/lib.rs:8:7
  |
8 |     F.foo::<u32>(S);
  |       ^^^------- help: remove the unnecessary generics
  |       |
  |       expected 0 generic arguments
  |
note: method defined here, with 0 generic parameters
 --> src/lib.rs:4:8
  |
4 |     fn foo(self, _x: impl Sized) {}
  |        ^^^
  = note: `impl Trait` cannot be explicitly specified as a generic argument

error[E0308]: mismatched types
 --> src/lib.rs:8:18
  |
8 |     F.foo::<u32>(S);
  |       ---------- ^ expected `u32`, found `S`
  |       |
  |       arguments to this method are incorrect
  |
note: method defined here
 --> src/lib.rs:4:8
  |
4 |     fn foo(self, _x: impl Sized) {}
  |        ^^^       --------------
```

### Desired output

```Shell
error[E0107]: method takes 0 generic arguments but 1 generic argument was supplied
 --> src/lib.rs:8:7
  |
8 |     F.foo::<u32>(S);
  |       ^^^------- help: remove the unnecessary generics
  |       |
  |       expected 0 generic arguments
  |
note: method defined here, with 0 generic parameters
 --> src/lib.rs:4:8
  |
4 |     fn foo(self, _x: impl Sized) {}
  |        ^^^
  = note: `impl Trait` cannot be explicitly specified as a generic argument
```

### Rationale and extra context

The second error is confusing, since rustc considers it as if we could specify the value of `impl Trait`, which we cannot. It would be better if the `impl Trait` type would be inferred.

### Other cases

```Rust

```

### Rust Version

```Shell
rustc 1.85.0 (4d91de4e4 2025-02-17)
```
